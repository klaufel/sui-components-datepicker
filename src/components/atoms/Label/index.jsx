import React from 'react'
import PropTypes from 'prop-types'
import './index.scss'

const Label = ({text, name}) => (
  <label className="Label" htmlFor={name}>
    {text}
  </label>
)

Label.propTypes = {
  /** The label itself */
  text: PropTypes.string.isRequired,
  /** Used as for attribute. Must be the same as the form element id */
  name: PropTypes.string.isRequired
}

export default Label
