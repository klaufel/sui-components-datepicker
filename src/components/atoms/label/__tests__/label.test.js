import React from 'react'
import {render} from '@testing-library/react'
import Label from '..'

describe('Label', () => {
  test('it works default', () => {
    const {container} = render(
      <Label text="Default label" name="default-label" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
