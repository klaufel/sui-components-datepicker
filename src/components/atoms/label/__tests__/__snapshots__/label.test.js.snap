// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label it works default 1`] = `
<label
  class="Label"
  for="default-label"
>
  Default label
</label>
`;
